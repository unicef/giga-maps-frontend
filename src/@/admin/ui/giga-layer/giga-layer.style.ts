import styled, { css } from "styled-components"

import { Scroll } from "~/@/scroll"

export const ViewLayerWrapper = styled.div`
  padding: 1rem;
 
`
export const GigaLayerScroll = styled(Scroll)`
 max-height: calc(100vh - 11.2rem);
`
export const LayerHeadingWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  padding-bottom: 1rem;
  a {
    font-size: 0.875rem;
    color: #277AFF;
  }
`

export const LayerContentWrapper = styled.div<{ top?: number }>`
  ${({ top = 0 }) => css`
      padding-top: ${top}rem;
  `}
  padding-bottom: 1rem;
  
`

export const LayerLabel = styled.p`
  color: #525252;
  font-size: 0.75rem;
`

export const LayerDetail = styled.p<{ $textTransform?: string; }>`
  margin-top: 0.5rem;
  color: #161616;
  font-size: 0.75rem;
  font-weight: 600;
  text-transform: ${(props) => props.$textTransform ?? 'none'};
`