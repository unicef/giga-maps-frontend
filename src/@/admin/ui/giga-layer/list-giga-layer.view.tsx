import { Add } from '@carbon/icons-react';
import { Button, Table, TableHeader, TableRow, TableToolbar, ToggletipButton } from '@carbon/react'
import { useStore } from 'effector-react';
import { useEffect, useState } from 'react'

import { $countryList } from '~/@/api-docs/models/explore-api.model';
import { StatusWrapper } from '~/@/api-docs/ui/components/api-keys-right-section/api-keys-right.side.style';
import { $userPermissions } from '~/core/auth/models';
import { addGigaLayer, viewGigaLayer } from '~/core/routes';
import { Link } from '~/lib/router';
import { DataSourceStatusChoices, DataSourceStatusNames } from '../../constants/giga-layer.constant';
import { $dataLayerListResponce, $dataListLayerCount, onGetDataLayerList } from '../../models/giga-layer.model';
import PageTitleComponent from '../common-components/page-title-component'
import Pagination from '../common-components/Pagination';
import { AdminTableScroll, CountryListToggletip, CountryListToggletipContent, DataLayerActiveCountries, SearchContainer, TableDataBody, TableDataCell, TableDataHead, TableWrapper } from '../styles/admin-styles'
import SearchToolbar from '../common-components/search-toolbar';

const ListGigaLayer = () => {
  const dataLayerList = useStore($dataLayerListResponce)
  const count = useStore($dataListLayerCount);
  const [{ page, pageSize }, setPageAndSize] = useState({ page: 1, pageSize: 20 });
  const [searchValue, setSearchValue] = useState('')
  const countryList = useStore($countryList)
  const userPermission = useStore($userPermissions);
  const isEditor = userPermission.CAN_ADD_DATA_LAYER;
  useEffect(() => {
    onGetDataLayerList({ page, pageSize, search: searchValue })
  }, [page, pageSize, searchValue])

  const getCountryName = (activeCountriesIds: number[]) => {
    const activeCountries = countryList.filter(item => activeCountriesIds.includes(item.id));
    return activeCountries
  }

  return (
    <>
      <PageTitleComponent
        title={"Giga layer"}
        subTitle={"List of giga layer"}
        recentlyView={false} />
      <TableToolbar>
        <SearchContainer>
          <SearchToolbar placeholder="Search by layer name, code" onSearchChange={setSearchValue} />
          <Link to={addGigaLayer}>
            <Button
              renderIcon={Add}
              onClick={() => { }}
              disabled={!isEditor}
            >
              Add new layer
            </Button>
          </Link>
        </SearchContainer>
      </TableToolbar>
      <AdminTableScroll $contentHeight="13.2rem">
        <TableWrapper>
          <Table aria-label="data-layer-list-data-table" >
            <TableDataHead >
              <TableRow>
                <TableHeader>
                  Unique Code
                </TableHeader>
                <TableHeader>
                  Giga Layer Label
                </TableHeader>
                <TableHeader>
                  Source API
                </TableHeader>
                <TableHeader>
                  Parameter
                </TableHeader>
                <TableHeader>
                  Source Ver
                </TableHeader>
                <TableHeader>
                  Status
                </TableHeader>
                <TableHeader>
                  Active Countries
                </TableHeader>
                <TableHeader>
                  Publisher
                </TableHeader>
              </TableRow>
            </TableDataHead>
            <TableDataBody>
              {
                dataLayerList?.map((item) => {
                  if (!item.created_by && item.type === 'LIVE') return null;
                  return (
                    <TableRow key={item.id}>
                      <TableDataCell>
                        <Link to={viewGigaLayer} params={{ id: item.id }}>{item?.code}</Link>
                      </TableDataCell>
                      <TableDataCell>
                        <Link to={viewGigaLayer} params={{ id: item.id }}>{item?.name} {!item.created_by && `(default)`}</Link>
                      </TableDataCell>
                      <TableDataCell>
                        {item?.data_sources_list.map((sourceApiObj, index, array) => (
                          <span key={sourceApiObj.id}>{sourceApiObj.name}
                            {index !== array.length - 1 && ','}
                          </span>
                        ))}
                      </TableDataCell>
                      <TableDataCell>
                        <span>{Object.values(item?.data_source_column ?? {})[0]?.alias}</span>
                      </TableDataCell>
                      <TableDataCell>

                        <span>{item?.data_sources_list[0].version}</span>

                      </TableDataCell>
                      <TableDataCell>
                        {
                          <StatusWrapper color={DataSourceStatusChoices[item?.status.toUpperCase()]}>
                            {DataSourceStatusNames[item?.status]}
                          </StatusWrapper>
                        }
                      </TableDataCell>
                      <TableDataCell>
                        {item?.applicable_countries.length > 0 ? item?.applicable_countries.length : countryList?.length}
                        <CountryListToggletip
                          align="bottom">
                          <ToggletipButton>
                            <DataLayerActiveCountries>
                              View list
                            </DataLayerActiveCountries>
                          </ToggletipButton>
                          <CountryListToggletipContent>
                            <p className="list-content">
                              {getCountryName(item.applicable_countries)?.map((country) => country?.name).join(', ') || 'All countries'}
                            </p>
                          </CountryListToggletipContent>
                        </CountryListToggletip>
                      </TableDataCell>
                      <TableDataCell>
                        {item?.published_by?.user_name ?? "--"}
                      </TableDataCell>
                    </TableRow>
                  )
                })
              }
            </TableDataBody>
          </Table >
        </TableWrapper>
      </AdminTableScroll>
      < Pagination
        page={page}
        pageSize={pageSize}
        count={count}
        setPage={setPageAndSize}
      />
    </>
  )
}

export default ListGigaLayer