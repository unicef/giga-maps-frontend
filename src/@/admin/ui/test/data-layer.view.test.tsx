import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { createEvent } from "effector";

import { $loggedInUser } from "~/core/auth/models";
import { currentLayerMockWithDraftStaus, currentLayerMockWithReadyToPublishStaus, dataLayerlistMock, singleLayerMock } from "~/tests/data/admin-data-layer";
import { loggedInUser } from "~/tests/data/admin-main-data";
import { testWrapper } from "~/tests/jest-wrapper";

import { getDataPreviewFx } from "../../effects/giga-layer-fx";
import { $currentGigaLayerItem, $dataLayerListResponce, $previewData, resetPreviewData } from "../../models/giga-layer.model";
import DataLayerMainView from "../giga-layer";
import AddEditGigaLayer from "../giga-layer/add-edit-giga-layer.view";
import ListGigaLayer from "../giga-layer/list-giga-layer.view";
import PreviewGigaLayer from "../giga-layer/preview-giga-layer";
import AdminViewLayer from "../giga-layer/view-giga-layer.view";
import { addGigaLayer, adminGigaLayer, editGigaLayer, viewGigaLayer } from "~/core/routes";
import { fetchMockResponse } from "~/tests/fetchMock";
import { getAppConfigValues } from "../../models/admin-model";
import { getCountryList } from "~/@/api-docs/models/explore-api.model";

const setDataLayerListResponce = createEvent();
$dataLayerListResponce.on(setDataLayerListResponce, (_, payload) => payload)

const setCurrentGigaLayerItem = createEvent()
$currentGigaLayerItem.on(setCurrentGigaLayerItem, (_, payload) => payload)

const setPreviewData = createEvent();
$previewData.on(setPreviewData, (_, payload) => payload)

const setLoggedInUser = createEvent();
$loggedInUser.on(setLoggedInUser, (_, payload) => payload);

describe('DataLayerMainView', () => {

  beforeEach(() => {
    getAppConfigValues()
    getCountryList(false);
    fetchMock.mockResponse(fetchMockResponse)
  })
  test('render DataLayerMainView and take snapshot', () => {
    const { asFragment } = render(
      testWrapper(<DataLayerMainView />)
    );
    expect(asFragment()).toMatchSnapshot();
  })

  test('render DataLayerMainView: adminGigaLayer,addGigaLayer,editGigaLayer,viewGigaLayer', () => {
    adminGigaLayer.navigate();
    const { rerender } = render(
      testWrapper(<DataLayerMainView />)
    );
    expect(window.location.pathname).toBe('/admin/giga-layer');
    // container.update();
    addGigaLayer.navigate();
    rerender(testWrapper(<DataLayerMainView />))
    expect(window.location.pathname).toBe('/admin/giga-layer/create');

    editGigaLayer.navigate({ id: 1 });
    rerender(testWrapper(<DataLayerMainView />))
    expect(window.location.pathname).toBe('/admin/giga-layer/edit/1');

    viewGigaLayer.navigate({ id: 1 });
    rerender(testWrapper(<DataLayerMainView />))
    expect(window.location.pathname).toBe('/admin/giga-layer/view/1');
  })

  test('render ListGigaLayer ', () => {
    const { asFragment } = render(
      testWrapper(<ListGigaLayer />)
    );
    expect(asFragment()).toMatchSnapshot();

  })

  test('render PreviewGigaLayer', async () => {
    const { getByTestId } = render(<PreviewGigaLayer isPreviewAvailable={true} />)
    expect(screen.getByText('Click on the button to see the preview on the Map.')).toBeInTheDocument()
    const button = getByTestId('data-layer-preview');
    await fireEvent.click(button);
    expect(screen.getByText('Preview')).toBeInTheDocument()
  })

  test("render with getDataPreviewFx is in pending state", () => {
    resetPreviewData();
    const spy = jest.spyOn(getDataPreviewFx.pending, 'getState');
    spy.mockReturnValue(true);
    void waitFor(() => {
      render(<PreviewGigaLayer isPreviewAvailable={true} />);
    });
    expect(screen.getByText('Loading data...')).toBeInTheDocument();
    spy.mockRestore();
  })

  test('render PreviewGigaLayer with isPreviewAvailable=false', () => {
    render(<PreviewGigaLayer isPreviewAvailable={false} />)
    expect(screen.getByText('Preview not available.')).toBeInTheDocument()
  })

  test("render with getDataPreviewFx is in pending state", () => {
    const { getByTestId } = render(testWrapper(<PreviewGigaLayer isPreviewAvailable={true} />));
    const button = getByTestId('data-layer-preview');
    fireEvent.click(button);
    const spy = jest.spyOn(getDataPreviewFx.pending, 'getState');
    spy.mockReturnValue(true);
    render(<PreviewGigaLayer isPreviewAvailable={true} />);
    // expect(screen.getByText('Loading data...')).toBeInTheDocument();
    spy.mockRestore();
  })

  test("click on publish button ", () => {
    setLoggedInUser(loggedInUser as any)
    setCurrentGigaLayerItem(currentLayerMockWithDraftStaus as any)
    const { getByTestId } = render(testWrapper(<AdminViewLayer />));
    const button = getByTestId('giga-layer-ready-to-publish');
    fireEvent.click(button);
  })

  test("click on publish button ", () => {
    setLoggedInUser(loggedInUser as any)
    setCurrentGigaLayerItem(currentLayerMockWithReadyToPublishStaus as any)
    const handleClick = jest.fn();
    const { getByTestId } = render(testWrapper(<AdminViewLayer />));
    const button = getByTestId('giga-layer-publish');
    fireEvent.click(button);
  })

  test("submit Giga Layer form with static layer", async () => {
    const { container, debug, getByTestId } = render(testWrapper(<AddEditGigaLayer />));
    // upload icon;
    const file = new File(['icon'], 'icon.svg', { type: 'image/svg' });
    const fileInput = container.querySelector('#file-upload input') as HTMLInputElement;
    // empty file
    fireEvent.change(fileInput, { target: { files: '' } });
    fireEvent.change(fileInput, { target: { files: [] } });

    fireEvent.change(fileInput, { target: { files: [file] } });

    const name = container.querySelector('#layer-name') as HTMLInputElement;
    await fireEvent.change(name, { target: { value: 'Test' } });
    const description = container.querySelector('#layer-description') as HTMLInputElement;
    await fireEvent.change(description, { target: { value: 'Test' } });
    const layerype = container.querySelector('#layer-type-select') as HTMLSelectElement;
    await fireEvent.change(layerype, { target: { value: 'STATIC' } });
    const sourceType = container.querySelector('#source-type') as HTMLSelectElement;
    fireEvent.click(sourceType.querySelector('button') as Element);
    await fireEvent.click(sourceType.querySelector('.cds--list-box__menu-item') as Element);
    // await fireEvent.change(sourceType, { selectedItems: [DataSourceType.DAILY_CHECK_APP] });
    const apiSource = container.querySelector('#apiSource-select') as HTMLSelectElement;
    fireEvent.click(apiSource.querySelector('button') as Element);
    await fireEvent.click(apiSource.querySelector('.cds--list-box__menu-item') as Element);

    const parameter = container.querySelector('#parameter-select') as HTMLSelectElement;
    await fireEvent.change(parameter, { target: { value: 'connectivity_speed' } });

    const countries = container.querySelector('#country-select') as HTMLSelectElement;
    fireEvent.click(countries.querySelector('button') as Element);
    await fireEvent.click(countries.querySelector('.cds--list-box__menu-item') as Element);

    const legendName0 = container.querySelector('#legend-name-0') as HTMLInputElement;
    await fireEvent.change(legendName0, { target: { value: '5G/5G' } });
    const legendValue0 = container.querySelector('#legend-value-0') as HTMLInputElement;
    await fireEvent.change(legendValue0, { target: { value: 'Good' } });

    const legendName1 = container.querySelector('#legend-name-1') as HTMLInputElement;
    await fireEvent.change(legendName1, { target: { value: '5G/5G' } });
    const legendValue1 = container.querySelector('#legend-value-1') as HTMLInputElement;
    await fireEvent.change(legendValue1, { target: { value: 'Moderate' } });

    const legendName2 = container.querySelector('#legend-name-2') as HTMLInputElement;
    await fireEvent.change(legendName2, { target: { value: '5G/5G' } });
    const legendValue2 = container.querySelector('#legend-value-0') as HTMLInputElement;
    await fireEvent.change(legendValue2, { target: { value: 'Bad' } });

    const legendName3 = container.querySelector('#legend-name-3') as HTMLInputElement;
    await fireEvent.change(legendName3, { target: { value: '5G/5G' } });
    const legendValue3 = container.querySelector('#legend-value-3') as HTMLInputElement;
    await fireEvent.change(legendValue3, { target: { value: 'Good' } });
    //  update legends
    const submit = getByTestId('submit-giga-layer-form')
    await fireEvent.click(submit)
  })

  test("submit Giga Layer form with live layer", async () => {
    const { container, getByTestId } = render(testWrapper(<AddEditGigaLayer />));

    const name = container.querySelector('#layer-name') as HTMLInputElement;
    await fireEvent.change(name, { target: { value: 'Test' } });
    const description = container.querySelector('#layer-description') as HTMLInputElement;
    await fireEvent.change(description, { target: { value: 'Test' } });
    const layerype = container.querySelector('#layer-type-select') as HTMLSelectElement;
    await fireEvent.change(layerype, { target: { value: 'LIVE' } });
    const sourceType = container.querySelector('#source-type') as HTMLSelectElement;
    fireEvent.click(sourceType.querySelector('button') as Element);
    await fireEvent.click(sourceType.querySelector('.cds--list-box__menu-item') as Element);
    // await fireEvent.change(sourceType, { selectedItems: [DataSourceType.DAILY_CHECK_APP] });
    const apiSource = container.querySelector('#apiSource-select') as HTMLSelectElement;
    fireEvent.click(apiSource.querySelector('button') as Element);
    await fireEvent.click(apiSource.querySelector('.cds--list-box__menu-item') as Element);

    const parameter = container.querySelector('#parameter-select') as HTMLSelectElement;
    await fireEvent.change(parameter, { target: { value: 'connectivity_speed' } });

    const countries = container.querySelector('#country-select') as HTMLSelectElement;
    fireEvent.click(countries.querySelector('button') as Element);
    await fireEvent.click(countries.querySelector('.cds--list-box__menu-item') as Element);

    const selectUnit = container.querySelector('#unit-select') as HTMLSelectElement;
    await fireEvent.change(selectUnit, { target: { value: 'mbps' } });

    const benchmark = container.querySelector('#global-giga-benchmark') as HTMLInputElement;
    await fireEvent.change(benchmark, { target: { value: 200000 } });

    const submit = getByTestId('submit-giga-layer-form')
    await fireEvent.click(submit)
  })
})