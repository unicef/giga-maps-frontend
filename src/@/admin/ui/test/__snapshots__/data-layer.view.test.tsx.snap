// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataLayerMainView render DataLayerMainView and take snapshot 1`] = `<DocumentFragment />`;

exports[`DataLayerMainView render ListGigaLayer  1`] = `
<DocumentFragment>
  .c1 >h3 {
  color: #FFF;
  font-family: Open Sans;
  font-size: 1.5rem;
  font-style: normal;
  font-weight: 500;
  line-height: 1.75rem;
}

.c1 >p {
  color: #F5F5F5;
  font-family: Open Sans;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: 400;
  line-height: 1.125rem;
}

.c0 {
  padding: 3.75rem 2rem 1rem 0rem;
  background: #222;
  display: flex;
  align-items: center;
}

.c0 button {
  margin-right: 1rem;
}

.c0 button path:not([data-icon-path]):not([fill=none]),
.c0 button .cds--btn__icon {
  fill: #fff!important;
}

<div
    class="c0"
  >
    <span
      class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--icon-tooltip"
    >
      <div
        class="cds--tooltip-trigger__wrapper"
      >
        <button
          aria-labelledby="tooltip-:r1d:"
          class="cds--btn--icon-only cds--btn cds--btn--md cds--layout--size-md cds--btn--ghost"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="cds--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.7 12.3L2.9 8.5 15 8.5 15 7.5 2.9 7.5 6.7 3.7 6 3 1 8 6 13z"
            />
          </svg>
        </button>
      </div>
      <span
        aria-hidden="true"
        class="cds--popover"
        id="tooltip-:r1d:"
        role="tooltip"
      >
        <span
          class="cds--popover-content cds--tooltip-content"
        >
          Go Back
        </span>
        <span
          class="cds--popover-caret"
        />
      </span>
    </span>
    <div
      class="c1"
    >
      <h3>
        Giga layer
      </h3>
      <p>
        List of giga layer
      </p>
    </div>
  </div>
  .c0 {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: end;
}

.c0 .cds--search-magnifier svg {
  fill: #222;
}

.c2 .cds--search-magnifier-icon {
  fill: #222;
}

.c1 {
  display: flex;
  width: 100%;
}

<section
    aria-label="data table toolbar"
    class="cds--table-toolbar"
  >
    <div
      class="c0"
    >
      <form
        class="cds--form c1"
      >
        <div
          aria-label="Search by layer name, code"
          class="cds--search cds--search--lg c2 cds--toolbar-search-container-active cds--toolbar-search-container-expandable"
          role="search"
        >
          <div
            class="cds--search-magnifier"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class="cds--search-magnifier-icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5 C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
              />
            </svg>
          </div>
          <label
            class="cds--label"
            for="2"
            id="2-search"
          >
            Filter table
          </label>
          <input
            autocomplete="off"
            class="cds--search-input"
            id="2"
            placeholder="Search by layer name, code"
            role="searchbox"
            tabindex="0"
            type="text"
            value=""
          />
          <button
            aria-label="Clear search input"
            class="cds--search-close cds--search-close--hidden"
            title="Clear search input"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.4141 16L24 9.4141 22.5859 8 16 14.5859 9.4143 8 8 9.4141 14.5859 16 8 22.5859 9.4143 24 16 17.4141 22.5859 24 24 22.5859 17.4141 16z"
              />
            </svg>
          </button>
        </div>
        <span
          class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--icon-tooltip"
        >
          <div
            class="cds--tooltip-trigger__wrapper"
          >
            <button
              aria-labelledby="tooltip-:r1g:"
              class="cds--btn--icon-only cds--btn cds--btn--secondary"
              tooltiptext="Search"
              type="submit"
            >
              <svg
                aria-hidden="true"
                aria-label="Search"
                class="cds--btn__icon"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5 C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
                />
              </svg>
            </button>
          </div>
          <span
            aria-hidden="true"
            class="cds--popover"
            id="tooltip-:r1g:"
            role="tooltip"
          >
            <span
              class="cds--popover-content cds--tooltip-content"
            >
              Search
            </span>
            <span
              class="cds--popover-caret"
            />
          </span>
        </span>
      </form>
      <a
        href="/admin/giga-layer/create"
      >
        <button
          class="cds--btn cds--btn--primary cds--btn--disabled"
          disabled=""
          type="button"
        >
          Add new layer
          <svg
            aria-hidden="true"
            class="cds--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
            />
          </svg>
        </button>
      </a>
    </div>
  </section>
  .c4 {
  padding: 1rem 1rem 1.5rem 1rem;
}

.c3 >tr >td {
  background: #FFFFFF;
  color: #474747;
  font-family: Open Sans;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: 400;
  line-height: 1.25rem;
  letter-spacing: 0.01rem;
}

.c2 >tr >th {
  border-top: #E0E0E0;
  border-bottom: #E0E0E0;
  background: #F2F2F2;
}

.c1 {
  min-height: calc(100vh - 10.5rem);
}

.c1 .cds--data-table thead {
  position: sticky;
  top: 0;
  z-index: 1;
}

.c1 .cds--data-table-content {
  overflow-x: initial;
}

.c7 {
  margin-left: 0.5rem;
  color: #277AFF;
  font-size: 0.75rem;
  font-style: normal;
  font-weight: 400;
  line-height: 1rem;
  letter-spacing: 0.02rem;
  text-decoration-line: underline;
  cursor: pointer;
}

.c6 :focus {
  outline: none!important;
}

.c6 .cds--popover-content .cds--toggletip-content {
  padding-top: 0.5rem;
}

.c6 .cds--popover-caret {
  display: none!important;
}

.c8 .list-content {
  margin-top: 0.5rem;
  font-size: 0.75rem;
  font-style: normal;
  font-weight: 400;
  line-height: 1rem;
}

.c0 {
  max-height: calc(100vh - 13.2rem);
}

.c5 {
  color: #474747;
  font-family: Open Sans;
  font-style: normal;
  font-weight: 400;
  line-height: 1rem;
  background: #B9EFCB;
  border-radius: 62rem;
  width: max-content;
  height: 1.5rem;
  padding: 0.3rem 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
}

<div
    class="scrollbar-container c0 ps"
  >
    <div
      class="c1"
    >
      <div
        class="cds--data-table-content"
      >
        <table
          aria-label="data-layer-list-data-table"
          class="cds--data-table cds--data-table--lg"
        >
          <thead
            class="c2"
          >
            <tr>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Unique Code
                </div>
              </th>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Giga Layer Label
                </div>
              </th>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Source API
                </div>
              </th>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Parameter
                </div>
              </th>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Source Ver
                </div>
              </th>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Status
                </div>
              </th>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Active Countries
                </div>
              </th>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Publisher
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            aria-live="polite"
            class="c3"
          >
            <tr>
              <td
                class="c4"
              >
                <a
                  href="/admin/giga-layer/view/8"
                />
              </td>
              <td
                class="c4"
              >
                <a
                  href="/admin/giga-layer/view/8"
                >
                  School 
                </a>
              </td>
              <td
                class="c4"
              >
                <span>
                  Daily Check APP and MLab,
                </span>
                <span>
                  QoS
                </span>
              </td>
              <td
                class="c4"
              >
                <span>
                  Upload Speed
                </span>
              </td>
              <td
                class="c4"
              >
                <span>
                  V1.0
                </span>
              </td>
              <td
                class="c4"
              >
                <div
                  class="c5"
                  color="#B9EFCB"
                >
                  Published
                </div>
              </td>
              <td
                class="c4"
              >
                3
                <span
                  class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--bottom cds--toggletip c6"
                >
                  <button
                    aria-controls="id-:r1r:"
                    aria-expanded="false"
                    aria-label="Show information"
                    class="cds--toggletip-button"
                    type="button"
                  >
                    <span
                      class="c7"
                    >
                      View list
                    </span>
                  </button>
                  <span
                    class="cds--popover"
                    id="id-:r1r:"
                  >
                    <span
                      class="cds--popover-content c8"
                    >
                      <div
                        class="cds--toggletip-content"
                      >
                        <p
                          class="list-content"
                        >
                          Afghanistan, Åland Islands
                        </p>
                      </div>
                    </span>
                    <span
                      class="cds--popover-caret"
                    />
                  </span>
                </span>
              </td>
              <td
                class="c4"
              >
                Test Admin
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="ps__rail-x"
      style="left: 0px; top: 0px;"
    >
      <div
        class="ps__thumb-x"
        style="left: 0px; width: 0px;"
        tabindex="0"
      />
    </div>
    <div
      class="ps__rail-y"
      style="top: 0px; left: 0px;"
    >
      <div
        class="ps__thumb-y"
        style="top: 0px; height: 0px;"
        tabindex="0"
      />
    </div>
  </div>
  .c0 .cds--popover {
  display: none;
}

<div
    class="cds--pagination cds--pagination--md c0"
  >
    <div
      class="cds--pagination__left"
    >
      <label
        class="cds--pagination__text"
        for="cds-pagination-select-:r1s:"
        id="cds-pagination-select-:r1s:-count-label"
      >
        Items per page
      </label>
      <div
        class="cds--form-item cds--select__item-count"
      >
        <div
          class="cds--select cds--select--inline"
        >
          <div
            class="cds--select-input--inline__wrapper"
          >
            <div
              class="cds--select-input__wrapper"
            >
              <select
                class="cds--select-input"
                id="cds-pagination-select-:r1s:"
                title=""
              >
                <option
                  class="cds--select-option"
                  value="10"
                >
                  10
                </option>
                <option
                  class="cds--select-option"
                  value="20"
                >
                  20
                </option>
                <option
                  class="cds--select-option"
                  value="50"
                >
                  50
                </option>
                <option
                  class="cds--select-option"
                  value="100"
                >
                  100
                </option>
                <option
                  class="cds--select-option"
                  value="500"
                >
                  500
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="cds--select__arrow"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <span
        class="cds--pagination__text cds--pagination__items-count"
      >
        1–2 of 2 items
      </span>
    </div>
    <div
      class="cds--pagination__right"
    >
      <div
        class="cds--form-item cds--select__page-number"
      >
        <div
          class="cds--select cds--select--inline"
        >
          <label
            class="cds--label cds--visually-hidden"
            for="cds-pagination-select-:r1s:-right"
          >
            Page number, of 1 pages
          </label>
          <div
            class="cds--select-input--inline__wrapper"
          >
            <div
              class="cds--select-input__wrapper"
            >
              <select
                class="cds--select-input"
                id="cds-pagination-select-:r1s:-right"
                title=""
              >
                <option
                  class="cds--select-option"
                  value="1"
                >
                  1
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="cds--select__arrow"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <span
        class="cds--pagination__text"
      >
        of 1 page
      </span>
      <div
        class="cds--pagination__control-buttons"
      >
        <span
          class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--icon-tooltip cds--icon-tooltip--disabled"
        >
          <div
            class="cds--tooltip-trigger__wrapper"
          >
            <button
              aria-labelledby="tooltip-:r1t:"
              class="cds--btn--icon-only cds--pagination__button cds--pagination__button--backward cds--pagination__button--no-index cds--btn cds--btn--ghost cds--btn--disabled"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 24L10 16 20 8z"
                />
              </svg>
            </button>
          </div>
          <span
            aria-hidden="true"
            class="cds--popover"
            id="tooltip-:r1t:"
            role="tooltip"
          >
            <span
              class="cds--popover-content cds--tooltip-content"
            >
              Previous page
            </span>
            <span
              class="cds--popover-caret"
            />
          </span>
        </span>
        <span
          class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top-right cds--tooltip cds--icon-tooltip cds--icon-tooltip--disabled"
        >
          <div
            class="cds--tooltip-trigger__wrapper"
          >
            <button
              aria-labelledby="tooltip-:r1v:"
              class="cds--btn--icon-only cds--pagination__button cds--pagination__button--forward cds--pagination__button--no-index cds--btn cds--btn--ghost cds--btn--disabled"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 8L22 16 12 24z"
                />
              </svg>
            </button>
          </div>
          <span
            aria-hidden="true"
            class="cds--popover"
            id="tooltip-:r1v:"
            role="tooltip"
          >
            <span
              class="cds--popover-content cds--tooltip-content"
            >
              Next page
            </span>
            <span
              class="cds--popover-caret"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
